src/components/Presets/AddPreset.svelte

```diff
@@ -0,0 +1,76 @@
+<script lang="ts">
+  import { version } from '../../../package.json'
+  import { patchStore, presetsStore } from '../../stores'
+  import { type Patch, Visibility } from '$lib/patch'
+  import { savePreset } from '$lib/presets'
+  import { DEFAULT_CATEGORIES } from '$lib/presets/constants'
+
+  export let handleCancelClick: () => void
+
+  const handleSubmit = async (event: Event): Promise<void> => {
+    const formEl = event.target as HTMLFormElement
+    const data: FormData = new FormData(formEl)
+
+    const preset: Patch = {
+      creator: data.get('creator') as string,
+      favorite: false,
+      id: crypto.randomUUID(),
+      name: data.get('name') as string,
+      notes: data.get('notes') as string,
+      params: {
+        delayTime: 200,
+        feedback: 0,
+        mix: 50
+      },
+      // @ts-ignore: Type mismatch from form data
+      tags: (data.get('tags') ? data.get('tags')?.split(',')?.map((tag: string) => tag.trim()) : []) as string[],
+      version,
+      // @ts-ignore: Type mismatch from form data
+      visibility: data.get('visibility') as Visibility,
+    }
+
+    await savePreset(preset)
+
+    // Parse tags as categories from presets
+    const categories = [...DEFAULT_CATEGORIES]
+    presetsStore.update(state => {
+      state.presets.forEach(({ tags }) => tags.forEach((tag) => categories.push(tag)))
+      return {
+        ...state,
+        categories,
+        selectedPatch: preset.id,
+      }
+    })
+
+    // Load the new patch into the patchStore
+    patchStore.update(() => ({ ...preset }))
+
+    // Leave the edit view
+    handleCancelClick()
+  }
+</script>
+
+<form on:submit={handleSubmit}>
+  <label for="name" class="mb-1 text-xs">Name</label>
+  <input type="text" name="name" class="input input-bordered w-full mb-3" required />
```

Comment:
We may want to add `spellcheck="false"` to these inputs. Safari seems a bit aggressive about making suggestions and capitalizing inputs in the webview.

Reply:
ahh good call! yeah the spellcheck has been a little annoying in safari 😅 i'll update it 👍🏼